examples/twitter-filter/elasticsearch_consumer/consumer.bal

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerinax/kafka;
+import ballerina/lang.value;
+
+// Topic to which the filtered tweets are published.
+const TOPIC = "twitter-tweets";
+
+// The URL of the Elasticsearch instance to use for all your queries.
+string URL = "http://localhost:9200";
+
+// Basic authentication details to connect to the Elasticsearch instance.
+string username = "username";
+string password = "password";
+
+kafka:ConsumerConfiguration consumerConfigs = {
+    groupId: "elastic-search",
+    topics: [TOPIC],
+    pollingInterval: 1,
+    // Sets the `autoCommit` to `false` so that the records should be committed manually.
+    autoCommit: false
+};
+
+listener kafka:Listener kafkaListener = new (kafka:DEFAULT_URL, consumerConfigs);
+
+// Creates a new HTTP client for the Elasticsearch node.
+final http:Client twitterClient = check new (URL,
```

Comment:
I think the better name would be something like `elkClient` because it connects to Elasticsearch node not twitter.

Reply:
Also I think there is no need to make this a module level variable.

Reply:
This is made a global variable as this is used in the function `processKafkaRecord`(L82). `processKafkaRecord` is called inside the Kafka consumer(L70) service per each message received. Making this a local variable will have to create a new client per message received.